uxarray/grid.py

```diff
@@ -0,0 +1,272 @@
+"""uxarray grid module."""
+import math
+import xarray as xr
+import numpy as np
+from ast import Raise
+from logging import raiseExceptions
+
+from pathlib import PurePath
+import os
+
+
+class Grid:
+    """A class for uxarray grid object.
+
+    Examples
+    ========
+    import uxarray as ux
+    # open an exodus file with Grid object
+    mesh = ux.Grid("filename.g")
+
+    # save as ugrid file
+    mesh.saveas("outfile.ug")
+    """
+
+    # Import read/write methods from files in this folder
+    from ._exodus import read_exodus, write_exodus
+    from ._ugrid import read_ugrid, write_ugrid
+    from ._shpfile import read_shpfile
+    from ._scrip import read_scrip
+
+    def __init__(self, *args, **kwargs):
+        """Initialize grid variables, decide if loading happens via file, verts
+        or gridspec If loading from file, initialization happens via the
+        specified file.
+
+        Raises:
+            RuntimeError: File not found
+        """
+        # initialize possible variables
+        self.filepath = None
+        self.gridspec = None
+        self.vertices = None
+        self.islatlon = None
+        self.concave = None
+        self.meshFileType = None
+
+        # this xarray variable holds an existing external netcdf file with loaded with xarray
+        self.ext_ds = None
+
+        # internal uxarray representation of mesh stored in internal object in_ds
+        self.in_ds = xr.Dataset()
+
+        # determine initialization type - string signifies a file, numpy array signifies a list of verts
+        in_type = type(args[0])
+
+        # check if this is a valid file:
+        try:
+            if os.path.isfile(args[0]) == False and in_type is not np.ndarray:
+                print("File not found: ", args[0])
+                exit()
+            elif in_type is np.ndarray:
+                self.vertices = args[0]
+                self.from_vert()
+        except ValueError as e:
+            # initialize from vertices
+            if in_type is np.ndarray:
+                self.vertices = args[0]
+                self.from_vert()
```

Comment:
Should be able to rearrange with use of kwargs. Also, If an exception does need to be raised I suggest using:

```
raise FileNotFoundError("File not found: " + args[0])
```